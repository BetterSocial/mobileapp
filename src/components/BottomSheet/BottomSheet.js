import * as React from 'react';
import {StyleSheet, View} from 'react-native';
import PropTypes from 'prop-types';
import RBSheet from 'react-native-raw-bottom-sheet';

import {COLORS} from '../../utils/theme';
import dimen from '../../utils/dimen';

const BottomSheet = React.forwardRef((props, ref) => {
  const {pullBottom = false, keyboardAvoidingViewEnabled = false} = props;
  return (
    <RBSheet
      ref={ref}
      onOpen={() => (props.onOpen ? props.onOpen() : {})}
      closeOnDragDown={true}
      dragFromTopOnly={true}
      closeOnPressMask={props.closeOnPressMask}
      height={props.height ? props.height : 355}
      keyboardAvoidingViewEnabled={keyboardAvoidingViewEnabled}
      customStyles={{
        wrapper: {
          backgroundColor: COLORS.black80
        },
        container: styles.containerSheet(pullBottom),
        draggableIcon: styles.draggableIcon
      }}>
      <View style={{...styles.container, ...props.viewstyle}}>{props.children}</View>
    </RBSheet>
  );
});

BottomSheet.displayName = 'BottomSheet';

BottomSheet.propTypes = {
  height: PropTypes.number,
  viewstyle: PropTypes.object
};

export default BottomSheet;

const styles = StyleSheet.create({
  container: {
    paddingTop: dimen.normalizeDimen(10),
    paddingHorizontal: dimen.normalizeDimen(20),
    paddingBottom: dimen.normalizeDimen(20),
    backgroundColor: COLORS.almostBlack
  },
  containerSheet: (pullBottom) => ({
    borderTopRightRadius: 20,
    borderTopLeftRadius: 20,
    justifyContent: pullBottom ? 'flex-end' : 'flex-start',
    backgroundColor: COLORS.almostBlack
  }),
  draggableIcon: {
    backgroundColor: COLORS.gray110
  }
});
