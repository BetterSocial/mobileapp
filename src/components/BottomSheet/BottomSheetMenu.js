import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import RBSheet from 'react-native-raw-bottom-sheet';
import dimen from '../../utils/dimen';
import {normalizeFontSize} from '../../utils/fonts';
import Gap from '../Gap';
import {COLORS} from '../../utils/theme';

const BottomSheetMenu = ({refSheet, dataSheet = [], height = 234}) => {
  return (
    <RBSheet
      ref={refSheet}
      closeOnDragDown={true}
      closeOnPressMask={true}
      customStyles={{
        container: styles.rbsheetContainer(height),
        draggableIcon: styles.rbsheetDraggableIcon
      }}>
      {dataSheet.map((item) => (
        <TouchableOpacity
          key={item.id}
          style={item.id === 1 ? styles.rbsheetContent : styles.rbsheetContent2}
          onPress={item.onPress}>
          {item.icon}
          <Gap width={dimen.normalizeDimen(12)} />
          <Text style={[styles.textShare, item.style]}>{item.name}</Text>
        </TouchableOpacity>
      ))}
    </RBSheet>
  );
};

export default BottomSheetMenu;

const styles = StyleSheet.create({
  rbsheetContainer: (height) => ({
    backgroundColor: COLORS.white,
    borderTopLeftRadius: dimen.normalizeDimen(12),
    borderTopRightRadius: dimen.normalizeDimen(12),
    height: dimen.normalizeDimen(height)
  }),
  rbsheetDraggableIcon: {
    backgroundColor: COLORS.greySubtile1,
    height: dimen.normalizeDimen(4),
    width: dimen.normalizeDimen(60),
    marginTop: dimen.normalizeDimen(16),
    marginBottom: dimen.normalizeDimen(24)
  },
  rbsheetContent: {
    flexDirection: 'row',
    alignItems: 'center',
    borderTopWidth: 1,
    borderBottomWidth: 1,
    paddingVertical: dimen.normalizeDimen(14),
    paddingHorizontal: dimen.normalizeDimen(20),
    borderColor: COLORS.greySubtile2
  },
  rbsheetContent2: {
    flexDirection: 'row',
    alignItems: 'center',
    borderBottomWidth: 1,
    paddingVertical: dimen.normalizeDimen(14),
    paddingHorizontal: dimen.normalizeDimen(20),
    borderColor: COLORS.greySubtile2
  },
  textShare: {
    fontSize: normalizeFontSize(16),
    fontWeight: '400',
    lineHeight: dimen.normalizeDimen(24),
    textAlignVertical: 'center'
  }
});
