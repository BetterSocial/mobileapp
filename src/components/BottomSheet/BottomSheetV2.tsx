import React, {forwardRef, ReactNode, Ref} from 'react';
import {StyleSheet, View, ViewStyle} from 'react-native';

import RBSheet, {RBSheetProps} from 'react-native-raw-bottom-sheet';
import {COLORS} from '../../utils/theme';
import dimen from '../../utils/dimen';

type BottomSheetProps = RBSheetProps & {
  viewstyle?: ViewStyle;
  pullBottom?: boolean;
  keyboardAvoidingViewEnabled?: boolean;
  onOpen?: () => void;
  closeOnPressMask?: boolean;
  children?: ReactNode;
};

const BottomSheetV2 = forwardRef((props: BottomSheetProps, ref: Ref<RBSheet>) => {
  const {pullBottom = false, keyboardAvoidingViewEnabled = false} = props;
  return (
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    <RBSheet
      ref={ref}
      onOpen={() => (props.onOpen ? props.onOpen() : {})}
      closeOnDragDown={true}
      dragFromTopOnly={true}
      closeOnPressMask={props.closeOnPressMask}
      height={props.height ?? 355}
      keyboardAvoidingViewEnabled={keyboardAvoidingViewEnabled}
      customStyles={{
        container: {
          borderTopRightRadius: 12,
          borderTopLeftRadius: 12,
          justifyContent: pullBottom ? 'flex-end' : 'flex-start'
        },
        draggableIcon: styles.draggableIcon
      }}>
      <View style={{...styles.container, ...props.viewstyle}}>{props.children}</View>
    </RBSheet>
  );
});

BottomSheetV2.displayName = 'BottomSheet';

export default BottomSheetV2;

const styles = StyleSheet.create({
  container: {
    paddingTop: dimen.normalizeDimen(10),
    paddingHorizontal: dimen.normalizeDimen(20),
    paddingBottom: dimen.normalizeDimen(20)
  },
  draggableIcon: {
    backgroundColor: COLORS.gray100
  }
});
