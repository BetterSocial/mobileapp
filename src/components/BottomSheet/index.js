import BottomSheet from './BottomSheet';

export {BottomSheet};
