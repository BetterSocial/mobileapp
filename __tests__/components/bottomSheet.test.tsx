import * as React from 'react';
import {shallow, configure} from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import BottomSheet from '../../src/components/BottomSheet/BottomSheet';

configure({adapter: new Adapter()});
describe('component BottomSheet', () => {
  it('BottomSheet snapshot', () => {
    const myRef = {
      current: {
        open: () => {},
        close: () => {},
      },
    };
    const component = shallow(<BottomSheet ref={myRef} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('BottomSheet renders correctly', () => {
    // const myRef = useRef();
    const tree = renderer.create(<BottomSheet />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
