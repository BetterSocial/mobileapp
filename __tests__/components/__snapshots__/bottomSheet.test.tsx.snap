// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component BottomSheet BottomSheet renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "portrait-upside-down",
      "landscape",
      "landscape-left",
      "landscape-right",
    ]
  }
  transparent={true}
  visible={false}
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "#00000077",
            "flex": 1,
          },
          undefined,
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "opacity": 1,
        }
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "height": 0,
          "justifyContent": "flex-start",
          "overflow": "hidden",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ccc",
                "borderRadius": 5,
                "height": 5,
                "margin": 10,
                "width": 35,
              },
              Object {
                "backgroundColor": "#E0E0E0",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "paddingBottom": 38,
            "paddingHorizontal": 20,
            "paddingTop": 20,
          }
        }
      />
    </View>
  </View>
</Modal>
`;

exports[`component BottomSheet BottomSheet snapshot 1`] = `
<RBSheet
  animationType="none"
  closeDuration={200}
  closeOnDragDown={true}
  closeOnPressBack={true}
  closeOnPressMask={true}
  customStyles={
    Object {
      "container": Object {
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "justifyContent": "flex-start",
      },
      "draggableIcon": Object {
        "backgroundColor": "#E0E0E0",
      },
    }
  }
  dragFromTopOnly={false}
  height={260}
  keyboardAvoidingViewEnabled={false}
  minClosingHeight={0}
  onClose={null}
  onOpen={[Function]}
  openDuration={300}
>
  <View
    style={
      Object {
        "paddingBottom": 38,
        "paddingHorizontal": 20,
        "paddingTop": 20,
      }
    }
  />
</RBSheet>
`;
